### DISCUSSION TITLE
Cron Job

### DISCUSSION THREAD

**Discussion by Skippy-007** [djmaze/resticker#194] [Q&A]
Hi while doing a large offsite backup (B2) which takes multiple days I find that if the cron job is configured to perform a backup each day at a particular time i.e. 6 am, i have observed weird behaviour. It seems that the cron job does not check that there is another backup active and keep looping trying to perform a backup. The solution is to stop and restart the container or push the cron job out a few days out. 

Has anyone seen this behaviour and is there a way to make it check for existing backup process.

**Comment by djmaze** [comment]
You are right, this is a bug (or at least a non-present feature) in the cron job library. 

The question is, what would you expect to happen in these cases?

a) Skip the new backup if an old one is still running
a) Just delay the new backup to run _immediately_ after the old backup has finished

As a workaround, you can for now just slow down your backup schedule (so it is run i.e. every 2 days instead of every day). This is non-optimal, I know.

**Comment by Skippy-007** [comment]
Hi  thanks for the reply.

In my view I would expect to skip the new backup until the current one
finishes and keep the same cron schedule. Ie don't run a backup immediately
after current backup finishes.

As a workaround I did have to delay the cron job as per your suggestion.

I hope that makes sense.

**Reply by djmaze** [reply]
#201
